erpnext/accounts/doctype/gl_entry/gl_entry.py

```diff
@@ -177,6 +179,28 @@ def pl_must_have_cost_center(self):
 
 			frappe.throw(msg, title=_("Missing Cost Center"))
 
+	def validate_company_in_accounting_dimension(self):
+		doc_field = DocType("DocField")
+		accounting_dimension = DocType("Accounting Dimension")
+		query = (
+			frappe.qb.from_(accounting_dimension)
+			.select(accounting_dimension.document_type)
+			.join(doc_field)
+			.on(doc_field.parent == accounting_dimension.document_type)
+			.where(doc_field.fieldname == "company")
+		).run(as_list=True)
+
+		dimension_list = sum(query, ["Project"])
+		for dimension in dimension_list:
+			if dimension_value := self.get(frappe.scrub(dimension)):
+				company = frappe.get_cached_value(dimension, dimension_value, "company")
```

Comment:
The query on line 185 and this `get_cached_value` will be happening for every `GL Entry` creation to solve this error. Is it too much @ruthra-kumar ?

Reply:
Also GL Entry document creation happens after the document is submitted (`on_submit` controller method). Validating for basic stuff should not happen at this stage.

Instead of here
```
Sales Invoice -> validate -> submit -> GL Entry -> validate (throw if dimension is not from company)
```


Throw here
```
Sales Invoice -> validate(throw if dimension is not from company) -> submit -> GL Entry -> validate 
```